### PR TITLE
Add more Kornia augmentations

### PR DESCRIPTION
**Pull request by alckasoc**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -13,11 +13,12 @@ dependencies:
   - lightning=2.0.5 # due to dependency conflict Lightning Issue (#18027)
   - cudnn
   - pytorch
-  - kornia
   - torchvision
+  - torchdata
```
Comment (talmo) [inline environment.yml:17]: We stopped using `torchdata` as of #9.

File: environment_cpu.yml [hunk 2]
```diff
@@ -11,11 +11,12 @@ dependencies:
   - pydantic<=2.0
   - lightning=2.0.5 # due to dependency conflict Lightning Issue (#18027)
   - cpuonly
-  - kornia
   - torchvision
+  - torchdata
```
Comment (talmo) [inline environment_cpu.yml:15]: We stopped using `torchdata` as of #9.

File: environment_osx-arm64.yml [hunk 3]
```diff
@@ -10,11 +10,12 @@ dependencies:
   - pydantic<2.0
   - lightning=2.0.5 # due to dependency conflict Lightning Issue (#18027)
   - pytorch
-  - kornia
   - torchvision
+  - torchdata
```
Comment (talmo) [inline environment_osx-arm64.yml:14]: We stopped using `torchdata` as of #9.

File: pyproject.toml [hunk 4]
```diff
@@ -19,14 +19,15 @@ classifiers = [
 dependencies = [
     "torch>=2.0.0",
     "torchvision",
+    "torchdata",
```
Comment (talmo) [inline pyproject.toml:22]: We stopped using `torchdata` as of #9.

File: sleap_nn/data/augmentation.py [hunk 5]
```diff
@@ -1,7 +1,93 @@
-"""This module implements data pipeline blocks for augmentations."""
-from typing import Optional
-from torch.utils.data.datapipes.datapipe import IterDataPipe
-import kornia.augmentation as K
+"""This module implements data pipeline blocks for augmentation operations."""
+from typing import Tuple, Dict, Any, Optional, Union, Text
+import torch
+from torchdata.datapipes.iter import IterDataPipe
```
Comment (talmo) [inline sleap_nn/data/augmentation.py:4]: We stopped using `torchdata` as of #9.

File: sleap_nn/data/augmentation.py [hunk 6]
```diff
@@ -38,20 +146,101 @@ class KorniaAugmenter(IterDataPipe):
     def __init__(
         self,
         source_dp: IterDataPipe,
-        rotation: float = 15.0,
+        crop_hw: Tuple[int, int],
+        crop_p: float = 1.0,
```
Comment (talmo) [inline sleap_nn/data/augmentation.py:150]: Can we make random cropping optional?

**Review by talmo — CHANGES_REQUESTED** [review]

File: tests/data/test_augmentation.py [hunk 7]
```diff
@@ -1,22 +1,76 @@
...
+    p = Normalizer(p)
+    try:
+        p = KorniaAugmenter(
+            p,
+            crop_hw=(0, 0),
+            crop_p=1.0,
+        )
+        assert False, "KorniaAugmenter RandomCrop failed to catch the size value error."
+    except ValueError as e:
+        pass
```
Comment (talmo) [inline tests/data/test_augmentation.py:76]: Use [`pytest.raises`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises) to test for expected exceptions

**Review by talmo — CHANGES_REQUESTED** [review]

**Review by talmo — APPROVED** [review]

**Review by gitttt-1234 — APPROVED** [review]

**Pull request merged**